### ISSUE TITLE
"App registrations (Preview)" disappears from Azure after Step 2 [if you are using a B2C tenant]

### ISSUE THREAD

**Issue by pritchie99** [Azure-Samples/active-directory-dotnet-native-aspnetcore-v2#38]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> I have followed through Step 2 - adding the entries for the 2 apps to the "App registrations (Preview)" blade.  However now when I go back via to the Azure Active Directory blade either examine or edit these entries, that option displays initially as the page responds to that menu click, and then disappears leaving only the previous "App registrations" item. 

(If I switch to another directory where I have not used the preview functionality yet, this menu item is visible as expected.)

### Any log messages given by the failure
> none.  Menu item simply disappears.

### Expected/desired behavior
> Menu item to stay available once I start using it's functionality.

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)
MacOS - Chrome and Safari, Windows10 - Edge

### Versions
>

### Mention any other details that might be useful
I know this is not a bug with this project.  But how do I submit it to the correct people - all the pages mentioning this preview functionality have no contact details at all.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by jmprieur** [comment]
@pritchie99 : thanks for the heads-up.
I've asked the portal team if there are known issues.

**Comment by jmprieur** [comment]
Do you want to try the following link?: https://aka.ms/appregistrations

**Comment by pritchie99** [comment]
Thanks @jmprieur.  Yes that link works well for me on whichever is the currently selected directory.  This is a great work-around so I can continue working with this preview so is very much appreciated.

**Comment by jmprieur** [comment]
good to know @pritchie99.
@lnalepa @manoj-rath FYI

**Comment by manoj-rath** [comment]
@pritchie99 Can you please confirm if you are on a B2C tenant, when the App registrations (Preview) link is missing in the menu bar on the left?

**Comment by pritchie99** [comment]
@manoj-rath No, it is just a normal AD tenant.  However it is not the "default" tenant for the subscription if that makes a difference.  I thought I would try this demo in a seperate tenant to the default.  NB the option did appear in the menu for that tenant prior to working through this demo and while adding entries for the TodolistService and TodolistClient.  However subsequent to that,  when I go away and come back to the tenant, as the menu renders, both the App Registrations, and App Registrations (Preview) appear in the menu initially.  Then the Preview item is removed.

**Comment by danieldobalian** [comment]
Hey @pritchie99!

This is an interesting issue. We would love to dig a bit deeper. Would you mind reaching out directly? My email is on my Github profile.  Thanks in advance!

**Comment by mstelten** [comment]
@pritchie99 Thanks again for alerting us to this issue.  Can you do one more favor?  If you are able to repro this bug again, please click the Smiley face in the upper right corner (this is the feedback icon) and put a quick note that you are reproing this issue.  This will provide me with the sessionId for your session so I can look through the logs and identify the issue.  Thanks again :)

**Comment by pritchie99** [comment]
@mstelten I have just done this after reproducing the issue with comment as follows:-

@mstelten
Reproduced issue "App registrations (Preview)" disappears from Azure after Step 2. #38.

**Comment by pritchie99** [comment]
@danieldobalian Email sent.

**Comment by Hartmannsson** [comment]
I am experiencing the same behaviour.  The preview option is available under one of my subscriptions (default?) but shows briefly whilst loading the blade in another.
Interestingly I tried the aka.ms link and it just hangs with three dots.

**Comment by pritchie99** [comment]
For me, Manoj was correct in that I had made this tenant a b2c tenant a long time ago and forgotten about it.  The preview does not work with b2c at this time.  It's not immediately obvious in the UI if this is the case and although there is a way to confirm this by viewing the manifest for the directory, however I don't recall how we did this.  However I can confirm that when I go to the App Registrations and view All Apps, I can see the b2c extension app registration there with the display name "b2c-extensions-app. Do not modify. Used by AADB2C for storing user data." so if you have one of those, I would suggest yours is b2c as well.
